Amazon.IonDotnet/Timestamp.cs

```diff
@@ -32,7 +30,8 @@ public enum Precision : byte
             Month = 2,
             Day = 3,
             Minute = 4,
-            Second = 5
+            Second = 5,
+            FractionalSecond = 6
```

Comment:
In ion-java we removed the dedicated `FracitonalSecond` precision because it led to many bugs over the years. I wonder if we can do the same thing here. Can we make the fractional seconds field nullable and then check whether it is present when the precision is `Second`?

Reply:
Changed the logic. There was no need for null. I used 0 as the default and check against it (0 versus 0.0). PR updated.
(Specifically in this commit: https://github.com/amzn/ion-dotnet/pull/100/commits/6d8eafc60fd5502cab26a754d9919f906a160de4)